Sources/Core/Microsoft.StreamProcessing/Egress/TemporalArray/TemporalArrayEgressPipe.cs

```diff
@@ -60,6 +60,23 @@ public override void OnNext(StreamMessage<Empty, TPayload> batch)
                 batch.Free();
             }
 
+            public override void OnCompleted()
+            {
+                OnFlush();
+                this.observer.OnCompleted();
```

Comment:
Same here - should this call base.OnCompleted instead? That will be closer to the pattern elsewhere and make sure that if we make it disposable that cleanup happens.